pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -990,8 +990,21 @@ func TestReconciler(t *testing.T) {
 					Admitted(true).
 					Obj(),
 			},
-			wantWorkloads:   []kueue.Workload{},
-			workloadCmpOpts: defaultWorkloadCmpOpts,
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("test-group", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet("b990493b", 2).Request(corev1.ResourceCPU, "1").Obj()).
+					Queue("test-queue").
+					ReserveQuota(
+						utiltesting.MakeAdmission("cq").
+							Assignment(corev1.ResourceCPU, "unit-test-flavor", "2").
+							AssignmentPodCount(2).
+							Obj(),
+					).
+					Admitted(true).
+					Delete().
```

Comment:
Should we verify in https://github.com/kubernetes-sigs/kueue/blob/6242c0ff8ab90887424805321d2275ff14a6765d/pkg/controller/jobs/pod/pod_controller_test.go#L2153 if the workload has `deletionTimestamp` instead of adding such a deletionTimestamp?